### DISCUSSION TITLE
How to get a folder relative to meson root folder, as an absolute path for find_library?

### DISCUSSION THREAD

**Discussion by aleinon** [mesonbuild/meson#10426] [Q&A]
Working in Windows 10, suppose I have a folder next to my meson project folder, which has a library I need, like this:

```
C:/Projects/MyMesonProject
C:/Projects/MyLibrary
```

Then in meson I get the library using the compiler like this:

```
cc = meson.get_compiler('cpp')
cc.find_library('mylib', dirs : ['C:/Projects/MyLibrary/lib'])
```

This works, but how can I change this so, that it works regardless of what path is in place of `C:/Projects`, considering that find_library requires an absolute path? So I need something like

`cc.find_library('mylib', dirs : [dirs : [meson.global_source_root() + '../MyLibrary/lib'])`

but this does not work with the `..`

What would be the correct way to accomplish this?

EDIT:

Answering to myself, need one more slash before the two dots like this:
 `cc.find_library('mylib', dirs : [dirs : [meson.global_source_root() + '/../MyLibrary/lib'])`
 
 Well, let's hope this will be helpful for someone.

**Comment by QuLogic** [comment]
Your meson build should definitely _definitely_ not assume some structure outside of it. Perhaps you might want to look into [subprojects](https://mesonbuild.com/Subprojects.html)?

**Reply by aleinon** [reply]
Definitely it's best to look into all options and choose what works the best for a particular use case. But subprojects deserve a mention in this context. Surely they are useful for many.
